### DISCUSSION TITLE
[make-fetch-happen] omits options passed through to minipass-fetch

### DISCUSSION THREAD

**Discussion by jmpage** [DefinitelyTyped/DefinitelyTyped#62216] [Issues with a @types package]
The `make-fetch-happen` library passes options ([1], [2], [3]) through to `minipass-fetch` (typed as `node-fetch`, which it reimplements). All options made available by `node-fetch` are supported by `minipass-fetch` ([4], [5]).

However, the typedef for `make-fetch-happen` only supports `RequestInit` options documented directly in its [README][6]:

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/5c248b832935aebece0abff07360102f1e9111f3/types/make-fetch-happen/index.d.ts#L23-L26

Would it be reasonable to make a PR to remove this restriction and allow the full set of `RequestInit` options supported by `minipass-fetch` to be provided to `make-fetch-happen`?

[1]: https://github.com/npm/make-fetch-happen/blob/2631db8306864f1812278d560b9bd0ca400478cd/lib/fetch.js#L96-L99
[2]: https://github.com/npm/make-fetch-happen/blob/2631db8306864f1812278d560b9bd0ca400478cd/lib/remote.js#L42-L46
[3]: https://github.com/npm/make-fetch-happen/blob/2631db8306864f1812278d560b9bd0ca400478cd/lib/remote.js#L51
[4]: https://github.com/npm/minipass-fetch/blob/e08e7d0fa1db87a272ef6df5c881988ec5bb4bbd/README.md#api
[5]: https://github.com/node-fetch/node-fetch/blob/6f72caa401a8ec574a22058431599ef47c222770/README.md
[6]: https://github.com/npm/make-fetch-happen/blob/2631db8306864f1812278d560b9bd0ca400478cd/README.md#--minipass-fetch-options

[a comment by an automated account is not shown]

**Comment by abernix** [comment]
That sound like a reasonable consideration, @jmpage!

When the choice was made originally, I think we were just opting to limit the need-for-investigation and reduce the scope of what we needed to evaluate type compatibility for.  In particular, the items that `make-fetch-happen` suggests [that it "augments"](https://github.com/npm/make-fetch-happen/blob/2631db8306864f1812278d560b9bd0ca400478cd/README.md#--make-fetch-happen-options) are probably of the most concern?

If you were to evaluate type compatibility to make sure it's compatible with the augmentation, then I'd be fully supportive of a PR that relaxes the narrowing of the type.  And in fact, even if we had to revisit this afterward in-light of inadvertent restrictions, that seems like more than reasonable as an approach for iterating on this.

Thoughts?

**Reply by jmpage** [reply]
Thank you for your reply, @abernix! I would be happy to take that approach. As for evaluating type compatibility, what do you have in mind:

Would it be reasonable to treat minipass-fetch as 1:1 compatible with the types of node-fetch, per the former's documentation? If so, would it be adequate to document how arguments are passed through from make-fetch-happen to minipass-fetch or are you thinking of something else?
